tests/unittests/core/test_apply_func.py

```diff
@@ -0,0 +1,326 @@
+import dataclasses
+import numbers
+from collections import defaultdict, namedtuple, OrderedDict
+from dataclasses import InitVar
+from typing import Any, ClassVar, List, Optional
+
+import pytest
+from unittests.core.mocks import torch
```

Comment:
In the origina, this also tests `np.ndarrays`. I replaced them with Tensors. I don't think that testing them explicitly is relevant if we already cover Tensors